pandas/core/internals/array_manager.py

```diff
@@ -400,7 +478,10 @@ def apply_with_block(self: T, f, align_keys=None, **kwargs) -> T:
                         # The caller is responsible for ensuring that
                         #  obj.axes[-1].equals(self.items)
                         if obj.ndim == 1:
-                            kwargs[k] = obj.iloc[[i]]
+                            if self.ndim == 2:
+                                kwargs[k] = obj.iloc[[i]]._values
```

Comment:
maybe `obj.iloc[slice(i, i+1)]` to avoid making a copy?

Reply:
Good idea, done